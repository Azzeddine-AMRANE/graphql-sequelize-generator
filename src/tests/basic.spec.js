const request = require('supertest')
const {
  migrateDatabase,
  seedDatabase,
  deleteTables
} = require('./testDatabase.js')
const express = require('express')
const { graphqlHttpServer } = require('./schema')
const http = require('spdy')

let app = null
let server = null

var options = {
  spdy: {
    plain: true
  }
}

/**
 * Starting the tests
 */
describe('Test the API queries', () => {
  beforeAll(async () => {
    app = express()
    app.use('/graphql', graphqlHttpServer)
    server = await new Promise((resolve, reject) => {
      const newServer = http
        .createServer(options, app)
        .listen(process.env.PORT || 8080, () => {
          resolve(newServer)
        })
    })
  })

  afterAll(() =>
    Promise.all([
      new Promise((resolve, reject) => server.close(() => resolve()))
    ])
  )

  beforeEach(async () => {
    try {
      await migrateDatabase()
      await seedDatabase()
    } catch (e) {
      console.log('Critical error during the database migration', e.message, e)
      throw e
    }
  })

  afterEach(async () => {
    await deleteTables()
  })

  it('Check that you can query a model and an association', async () => {
    const response = await request(server).get(
      `/graphql?query=
          query getCompanies {
            organizations: company {
              id
              name
              users {
                id
                name
              }
            }
          }
          &operationName=getCompanies`
    )
    const organizations = response.body.data.organizations
    expect(organizations).toMatchSnapshot('All companies')
  })
})
