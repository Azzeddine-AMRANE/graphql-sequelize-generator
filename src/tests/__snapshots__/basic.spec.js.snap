// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the API queries Check that you can query a model and an association: All companies 1`] = `
Array [
  Object {
    "id": 1,
    "name": "Company 1",
    "users": Array [
      Object {
        "id": 5,
        "name": "Test 5 c 2",
      },
      Object {
        "id": 10,
        "name": "Test 10 c 2",
      },
      Object {
        "id": 15,
        "name": "Test 15 c 2",
      },
      Object {
        "id": 20,
        "name": "Test 20 c 2",
      },
      Object {
        "id": 25,
        "name": "Test 25 c 2",
      },
      Object {
        "id": 30,
        "name": "Test 30 c 2",
      },
      Object {
        "id": 35,
        "name": "Test 35 c 2",
      },
      Object {
        "id": 40,
        "name": "Test 40 c 2",
      },
      Object {
        "id": 45,
        "name": "Test 45 c 2",
      },
      Object {
        "id": 50,
        "name": "Test 50 c 2",
      },
      Object {
        "id": 55,
        "name": "Test 55 c 2",
      },
      Object {
        "id": 60,
        "name": "Test 60 c 2",
      },
      Object {
        "id": 65,
        "name": "Test 65 c 2",
      },
      Object {
        "id": 70,
        "name": "Test 70 c 2",
      },
      Object {
        "id": 75,
        "name": "Test 75 c 2",
      },
      Object {
        "id": 80,
        "name": "Test 80 c 2",
      },
      Object {
        "id": 85,
        "name": "Test 85 c 2",
      },
      Object {
        "id": 90,
        "name": "Test 90 c 2",
      },
      Object {
        "id": 95,
        "name": "Test 95 c 2",
      },
      Object {
        "id": 100,
        "name": "Test 100 c 2",
      },
      Object {
        "id": 105,
        "name": "Test 105 c 2",
      },
      Object {
        "id": 110,
        "name": "Test 110 c 2",
      },
      Object {
        "id": 115,
        "name": "Test 115 c 2",
      },
      Object {
        "id": 120,
        "name": "Test 120 c 2",
      },
      Object {
        "id": 125,
        "name": "Test 125 c 2",
      },
      Object {
        "id": 130,
        "name": "Test 130 c 2",
      },
      Object {
        "id": 135,
        "name": "Test 135 c 2",
      },
      Object {
        "id": 140,
        "name": "Test 140 c 2",
      },
      Object {
        "id": 145,
        "name": "Test 145 c 2",
      },
      Object {
        "id": 150,
        "name": "Test 150 c 2",
      },
      Object {
        "id": 155,
        "name": "Test 155 c 2",
      },
      Object {
        "id": 160,
        "name": "Test 160 c 2",
      },
      Object {
        "id": 165,
        "name": "Test 165 c 2",
      },
      Object {
        "id": 170,
        "name": "Test 170 c 2",
      },
      Object {
        "id": 175,
        "name": "Test 175 c 2",
      },
      Object {
        "id": 180,
        "name": "Test 180 c 2",
      },
      Object {
        "id": 185,
        "name": "Test 185 c 2",
      },
      Object {
        "id": 190,
        "name": "Test 190 c 2",
      },
      Object {
        "id": 195,
        "name": "Test 195 c 2",
      },
      Object {
        "id": 200,
        "name": "Test 200 c 2",
      },
      Object {
        "id": 205,
        "name": "Test 205 c 2",
      },
      Object {
        "id": 210,
        "name": "Test 210 c 2",
      },
      Object {
        "id": 215,
        "name": "Test 215 c 2",
      },
      Object {
        "id": 220,
        "name": "Test 220 c 2",
      },
      Object {
        "id": 225,
        "name": "Test 225 c 2",
      },
      Object {
        "id": 230,
        "name": "Test 230 c 2",
      },
      Object {
        "id": 235,
        "name": "Test 235 c 2",
      },
      Object {
        "id": 240,
        "name": "Test 240 c 2",
      },
      Object {
        "id": 245,
        "name": "Test 245 c 2",
      },
      Object {
        "id": 250,
        "name": "Test 250 c 2",
      },
    ],
  },
  Object {
    "id": 3,
    "name": "Company 3",
    "users": Array [],
  },
  Object {
    "id": 5,
    "name": "Company 5",
    "users": Array [],
  },
  Object {
    "id": 7,
    "name": "Company 7",
    "users": Array [],
  },
]
`;

exports[`Test the API queries Check that you can query sub associations: All companies with users and their department 1`] = `
Array [
  Object {
    "id": 1,
    "name": "Company 1",
    "users": Array [
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 5,
        "name": "Test 5 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 10,
        "name": "Test 10 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 15,
        "name": "Test 15 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 20,
        "name": "Test 20 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 25,
        "name": "Test 25 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 30,
        "name": "Test 30 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 35,
        "name": "Test 35 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 40,
        "name": "Test 40 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 45,
        "name": "Test 45 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 50,
        "name": "Test 50 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 55,
        "name": "Test 55 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 60,
        "name": "Test 60 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 65,
        "name": "Test 65 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 70,
        "name": "Test 70 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 75,
        "name": "Test 75 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 80,
        "name": "Test 80 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 85,
        "name": "Test 85 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 90,
        "name": "Test 90 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 95,
        "name": "Test 95 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 100,
        "name": "Test 100 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 105,
        "name": "Test 105 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 110,
        "name": "Test 110 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 115,
        "name": "Test 115 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 120,
        "name": "Test 120 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 125,
        "name": "Test 125 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 130,
        "name": "Test 130 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 135,
        "name": "Test 135 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 140,
        "name": "Test 140 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 145,
        "name": "Test 145 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 150,
        "name": "Test 150 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 155,
        "name": "Test 155 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 160,
        "name": "Test 160 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 165,
        "name": "Test 165 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 170,
        "name": "Test 170 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 175,
        "name": "Test 175 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 180,
        "name": "Test 180 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 185,
        "name": "Test 185 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 190,
        "name": "Test 190 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 195,
        "name": "Test 195 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 200,
        "name": "Test 200 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 205,
        "name": "Test 205 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 210,
        "name": "Test 210 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 215,
        "name": "Test 215 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 220,
        "name": "Test 220 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 225,
        "name": "Test 225 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 230,
        "name": "Test 230 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 235,
        "name": "Test 235 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 240,
        "name": "Test 240 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 245,
        "name": "Test 245 c 2",
      },
      Object {
        "department": Object {
          "id": 1,
        },
        "id": 250,
        "name": "Test 250 c 2",
      },
    ],
  },
  Object {
    "id": 3,
    "name": "Company 3",
    "users": Array [],
  },
  Object {
    "id": 5,
    "name": "Company 5",
    "users": Array [],
  },
  Object {
    "id": 7,
    "name": "Company 7",
    "users": Array [],
  },
]
`;

exports[`Test the API queries Check that you can sort: All companies sorted by name 1`] = `
Array [
  Object {
    "id": 1,
    "name": "Company 1",
  },
  Object {
    "id": 3,
    "name": "Company 3",
  },
  Object {
    "id": 5,
    "name": "Company 5",
  },
  Object {
    "id": 7,
    "name": "Company 7",
  },
]
`;

exports[`Test the API queries One can exclude a basic field from a model 1`] = `
Array [
  Object {
    "extensions": Object {
      "code": "GRAPHQL_VALIDATION_FAILED",
    },
    "locations": Array [
      Object {
        "column": 15,
        "line": 6,
      },
    ],
    "message": "Cannot query field \\"updatedAt\\" on type \\"department\\". Did you mean \\"createdAt\\"?",
  },
]
`;

exports[`Test the API queries One can exclude an associated model 1`] = `
Array [
  Object {
    "extensions": Object {
      "code": "GRAPHQL_VALIDATION_FAILED",
    },
    "locations": Array [
      Object {
        "column": 15,
        "line": 6,
      },
    ],
    "message": "Cannot query field \\"company\\" on type \\"department\\". Did you mean \\"companyId\\"?",
  },
]
`;
